libs/map/src/lib/vl-map.ts

```diff
@@ -146,6 +149,32 @@ export class VlMap extends BaseElementOfType(HTMLElement) {
         this.__updateOverviewMapSizeOnLoad();
 
         this._map.addControl(this.__createZoomControl());
+
+        this.observeRemovedMapLayers();
+    }
+
+    disconnectedCallback(): void {
+        if (this.observer) {
+            this.observer.disconnect();
+        }
+    }
+
+    private observeRemovedMapLayers(): void {
+        const mapElement = this as unknown as HTMLElement;
+        this.observer = new MutationObserver((mutations: MutationRecord[]) => {
+            // opbouwen lijst van alle VlMapLayer-nodes die verwijderd werden uit deze instantie van het VlMap component
+            const removedMapNodes: Node[] = mutations
+                .filter(({ target }) => target === mapElement)
+                .map(({ removedNodes }) => Array.from(removedNodes).filter((node) => node instanceof VlMapLayer))
+                .reduce((allRemovedNodes, removedNodes) => allRemovedNodes.concat(removedNodes), [])
+                .filter((removedNode) => removedNode instanceof VlMapLayer);
+            const removedMapLayers = removedMapNodes as unknown[] as VlMapLayer[];
+            for (const removedMapLayer of removedMapLayers) {
+                // voor elke map layer uit te dom verwijderd, verwijder deze ook van de OL OverlayLayerCollection
+                this.map.removeOverlayLayer((<VlMapLayer>removedMapLayer)._layer);
+            }
```

Comment:
Ik denk dat je deze for-loop kan veranderen naar een forEach() en deze verder chainen op het vorige.

Reply:
Ja kan zeker, is een gewoonte van vroeger. Normaal gezien naar performance is een for-loop iets performanter, maar dit is niet relevant in dit geval & leesbaarheid is belangrijker! ik pas het aan